### PR TITLE
FIX for python restapi and WIP for publisher

### PR DESCRIPTION
**Pull request by manisabri**

### CODE REVIEW

**Comment by manisabri** [comment]
Test are not passing. In python `TruthValue` expects an `opencog.atomspace.PtrHolder` but we call it with  two numbers like `(0.1, 0.1) `
 Is this a known issue?

**Comment by linas** [comment]
Thanks! post-merge comments to follow ...

File: CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,10 @@ ENDIF(CMAKE_VERSION VERSION_GREATER 3.0.2)
 
 PROJECT(opencog)
 
+FIND_PACKAGE(Boost 1.71 REQUIRED date_time program_options thread filesystem system unit_test_framework)
```
Comment (linas) [inline CMakeLists.txt:24]: could you change this to 1.67 or lower? Debian stable runs on 1.67...

File: tests/persist/zmq/events/CMakeLists.txt [hunk 2]
```diff
@@ -13,6 +13,9 @@ LINK_LIBRARIES(
 	${ATOMSPACE_LIBRARIES}
 	zmq
 	tbb
+	attention-types
+    attention
+    atomspacepublishermodule
 )
```
Comment (linas) [inline tests/persist/zmq/events/CMakeLists.txt:19]: indentation

**Review by linas — APPROVED** [review]
minor comments

**Comment by linas** [comment]
> TruthValue

The 6 different kinds of TruthValues are all special cases of Values, in general. The RESTapi was created when there were only two or 3 different kinds of TruthValues, so, sure it probably has an inconsistent API.  It should be updated to handle Values in general.  Here's a quick recap:

There are three generic kinds of Values: those that hold a vector of floats, a vector of strings, and a vector of Values.  (Note that Atoms are a special case of Values) Both Atoms and Values are tracked with smart pointers; the smart-pointer for Atoms is called "Handle" and has some extra Atom-specific methods on it.

Most of the TruthValues are special cases of FloatValue, and hold 2 or more floats. Some of the TruthValues hold more complex structures (upper and lower bounds, a distribution, some other stats).

AttentionValues are special cases of FloatValue.  The AttentionValues are loaded only when the AttentionBank is loaded.

Values can encapsulate rapidly time-varying data, e.g. video audio streams, or more likely: 3D position, orientation, distances, angles, etc. and provide the latest values when queried. The spacetime values are loaded only when SpaceTime is loaded...

In other words, the collection of Values, just like the collection of Atoms, depends on which modules have been loaded. ...

> event publisher

The original even publisher is a horrible design. It uses callbacks to be notified of truth-value changes .. the problem is the callback craters performance, slowing down the system by 2x or 4x or something like that. and so is basically unusable/untenable in a running system. Some other kind of system is required.  Maybe some system that penalizes only those people who want this data, instead of penalizing everyone. Some kind of special custom server that sits on top of the AtomSpace, instead of inside of it ....

**Comment by manisabri** [comment]
So the next pull request gonna be for AtomSpace to fix the values I guess (in python bindings)

File: tests/persist/zmq/events/CMakeLists.txt (hunk 2 above)
Comment (manisabri) [reply tests/persist/zmq/events/CMakeLists.txt:19]: issued a new PR for the changes
